content/docs/instrumenting/http_sd.md

```diff
@@ -0,0 +1,93 @@
+---
+title: HTTP SD
+sort_rank: 7
+---
+
+# Writing HTTP Service Discovery
+
+Prometheus provides a generic [HTTP Service Discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config),
+that enables to discover targets over an HTTP endpoint.
+
+The HTTP Service Discovery is complimentary to the supported service
+discoveries, and is an alternative to the [File-based Service Discovery](https://prometheus.io/docs/guides/file-sd/#use-file-based-service-discovery-to-discover-scrape-targets).
+
+## Comparison between File-Based SD and HTTP SD
+
+Here is a table comparing our two generic Service Discovery implementations.
+
+| Item | File SD | HTTP SD |
+| ---- | ------- | ------- |
+| Event Based | Yes, via inotify | No |
+| Update frequency | Instant, thanks to inotify | Following refresh_interval |
+| Format | Yaml or JSON | JSON |
+| Transport | Local file | HTTP/HTTPS |
+| Security | File-Based security | TLS, Basic auth, Authorization header, OAuth2 |
+
+## Requirements of HTTP SD endpoints
+
+The HTTP SD endpoints should have been written specifically for Prometheus.
```

Comment:
That's unclear.

Reply:
What I try to say really is that we expect this to work only for endpoints crafted for this purpose. We don't aim to support e.g. file servers that would not provide appropriate content types etc...

Reply:
How can we better phrase it?